Shared/Extensions/String+PhoneNumbers.swift

```diff
@@ -27,3 +27,14 @@ extension String {
         return self.filter("0123456789".contains)
     }
 }
+
+/// A phone number  that is considerer equal if the last ten digits are the same as anothern fuzzy phone number.
+/// NOTE: This equality check can fail and may (rarely) result in false positives.
+struct FuzzyPhoneNumber: Hashable {
+
+    let partialNumber: Substring
+
+    init(_ phoneNumber: String) {
+        self.partialNumber = phoneNumber.removeAllNonNumbers().suffix(10)
+    }
+}
```

Comment:
Comparing phone numbers is actually really difficult because people often don't include the country code. See comment.